### DISCUSSION TITLE
评论挂件使用的问题

### DISCUSSION THREAD

**Discussion by KnightKilito** [walinejs/waline#1375] [Q&A] [answered]
我的博客是Hexo x Fluid

[闪闪の小窝](https://moechun.fun)

我前端没有好好学😭

![控制台的报错](https://user-images.githubusercontent.com/61006533/188803869-b8113192-3a58-40f8-b79c-35445f4f652a.png)
![js报错](https://user-images.githubusercontent.com/61006533/188803741-da0e9daa-ea98-4821-9a69-2adc25e4ba8d.png)

像这样的问题是不是因为博客主题的waline客户端是直接从cdn拿的然后找不到Waline这个对象
![cdn](https://user-images.githubusercontent.com/61006533/188804919-55cf1a7f-23d3-4e44-8fc8-f213eb3960e8.png)

如果我想用这个挂件的话该怎么做

**Comment by Mister-Hope** [comment]
你使用的这个主题它是自行导入 waline 的，而没有把 Waline 挂载在 windows 上。

**Reply by KnightKilito** [reply]
我去看看文档，想想看怎么搞

**Reply by KnightKilito** [reply]
今天打开突然就有了！看来其实是评论数据没有快速加载出来的原因

**Comment by Mister-Hope** [comment, marked as answer]
我没空去看主题代码，如果你的主题是现加载 Waline 的 CDN 版本，那么 windows onload 触发时，不能保证主题已经执行了评论相关代码并为你拉取了 Waline。你应该自行 modify 主题或者尝试寻找主题提供的评论加载完成的钩子来执行这一操作，当然你也可以自行引入主题使用的 Waline 链接然后调用 Waline。

**Reply by KnightKilito** [reply]
好滴，感谢您的思路😺
